homeassistant/components/synology_dsm/camera.py

```diff
@@ -86,10 +92,14 @@ def motion_detection_enabled(self):
 
     def camera_image(self) -> bytes:
         """Return bytes of camera image."""
+        if not self.available:
+            return None
         return self._api.surveillance_station.get_camera_image(self._camera.id)
 
     async def stream_source(self) -> str:
         """Return the source of the stream."""
+        if not self.available:
+            return None
         return self._camera.live_view.rtsp
```

Comment:
Checks can be added in `homeassistant.components.camera.Camera`, right ?

Reply:
Sure thing. Found an example in esphome:
https://github.com/home-assistant/core/blob/c8ac787c9f7be9059dfefdf937bcb270f87aa226/homeassistant/components/esphome/camera.py#L56-L59

Reply:
Yeah, that would be nice.

As this PR is for 0.116.x, I think I should do this in a new PR, for next release.